platform/broadcom/sonic-platform-modules-accton/as9716-32d/sonic_platform/fan.py

```diff
@@ -46,3 +46,9 @@ def get_direction(self):
 
         return direction
 
+    def get_target_speed(self):
+        if self.is_psu_fan:
+            raise NotImplementedError # Target speed not supported for PSU fans
```

Comment:
@seanwu-ec 
Do you think its better to implement this is the common pddf_fan.py ? 

# git diff  platform/pddf/platform-api-pddf-base/sonic_platform_pddf_base/pddf_fan.py
diff --git a/platform/pddf/platform-api-pddf-base/sonic_platform_pddf_base/pddf_fan.py b/platform/pddf/platform-api-pddf-base/sonic_platform_pddf_base/pddf_fan.py
index 30be0dd74..d272d40e6 100755
--- a/platform/pddf/platform-api-pddf-base/sonic_platform_pddf_base/pddf_fan.py
+++ b/platform/pddf/platform-api-pddf-base/sonic_platform_pddf_base/pddf_fan.py
@@ -224,7 +224,7 @@ class PddfFan(FanBase):
         target_speed = 0
         if self.is_psu_fan:
             # Target speed not usually supported for PSU fans
-            target_speed = 0
+            raise NotImplementedError
         else:
             idx = (self.fantray_index-1)*self.platform['num_fans_pertray'] + self.fan_index
             attr = "fan" + str(idx) + "_pwm"
#

Reply:
Yes, because PSU doesn't reveal target fan speed. So I think it's better pddf_fan.py implement this.
I've raised an issue for that sometime before: https://github.com/Azure/sonic-buildimage/issues/8129

In order to pass the fan status check at [https://github.com/Azure/sonic-platform-daemons/blob/master/sonic-thermalctld/scripts/thermalctld#L341](url), according to the try_get() in thermalctld, there are two ways of doing this. One is `raise NotImplementedError` another one is `return None`. You can go whichever you like.

Reply:
Thanks @seanwu-ec 
I added the fix in common PDDF APIs as part of PR (https://github.com/Azure/sonic-buildimage/pull/7834)

Reply:
You can remove those changes from individual platform's fan.py

Reply:
Done. 😊 Thanks,  @FuzailBrcm